tests/kafkatest/services/kafka/kafka.py

```diff
@@ -52,6 +53,107 @@ def listener_security_protocol(self):
         return "%s:%s" % (self.name, self.security_protocol)
 
 class KafkaService(KafkaPathResolverMixin, JmxMixin, Service):
+    """
+    Ducktape system test service for Brokers and Raft-based Controllers
+
+    Metadata Quorums
+    ----------------
+    Kafka can use either ZooKeeper or a Raft Controller quorum for its
+    metadata.  Raft Controllers can either be co-located with Kafka in
+    the same JVM or remote in separate JVMs.  The choice is made via
+    the 'metadata_quorum' parameter defined for the system test: if it
+    is not explicitly defined, or if it is set to 'ZK', then ZooKeeper
+    is used.  If it is explicitly set to 'INPROC_RAFT' then Raft
+    controllers will be co-located with the brokers; the value
+    `REMOTE_RAFT` indicates remote controllers.
+
+    You can interrogate the state of the instance via these attributes:
+
+    using_zk : bool
```

Comment:
If you make "quorum" a real object, then you can just have accessor methods on that type.  That seems much preferrable.

On a related note, we need some way to specify how many servers are controllers in the co-located case.  They shouldn't all be controllers.  (We have some very big tests with 10+ brokers)